### DISCUSSION TITLE
How to use BigDecimal for math calculations?

### DISCUSSION THREAD

**Discussion by shahamit** [projectnessie/cel-java#436] [Q&A]
We want to use this library for financial calculations and specifically use `BigDecimal` parameters to have accurate decimal calculations. I could not find `Decls.BigDecimal` type hence starting this discussion. 

How can I execute calculations on BigDecimal's?

Thanks.

**Comment by snazy** [comment]
CEL-Java, as all CEL implementations, are tied to the data types defined in the [CEL spec](https://github.com/google/cel-spec/blob/master/doc/langdef.md#numeric-values).

**Reply by shahamit** [reply]
Ok. That would mean that this library cannot be used for float point calculations (financial calculations) right? Am I missing something?

for e.g. the below code gives incorrect output as `0.19999999999999998`
```java
Script script = scriptHost.buildScript("x - y")
                .withDeclarations(
                        Decls.newVar("x", Decls.Double),
                        Decls.newVar("y", Decls.Double))
                .build();

        Map<String, Object> arguments = new HashMap<>();
        arguments.put("x", 0.3d);
        arguments.put("y", 0.1d);

        Double result = script.execute(Double.class, arguments);
```
